pkg/kubeapiserver/options/cloudprovider.go

```diff
@@ -49,42 +41,3 @@ func (s *CloudProviderOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.CloudConfigFile, "cloud-config", s.CloudConfigFile,
 		"The path to the cloud provider configuration file. Empty string for no configuration file.")
 }
-
-func (s *CloudProviderOptions) DefaultExternalHost(genericoptions *genericoptions.ServerRunOptions) error {
```

Comment:
removal of this behavior makes sense to deprecate, since there's an alternative (determine the hostname externally and pass in via flag)

before removing, we should 1) announce deprecation, and 2) update in tree deployers to obtain the host externally and pass it in via the command line instead of depending on this

Reply:
commented below